uaclient/clouds/identity.py

```diff
@@ -11,13 +13,32 @@
     # typing isn't available on trusty, so ignore its absence
     pass
 
+# Needed for Trusty-only
 CLOUDINIT_RESULT_FILE = "/var/lib/cloud/data/result.json"
+CLOUDINIT_INSTANCE_ID_FILE = "/var/lib/cloud/data/instance-id"
 
 
 # Mapping of datasource names to cloud-id responses. Trusty compat with Xenial+
 DATASOURCE_TO_CLOUD_ID = {"azurenet": "azure", "ec2": "aws"}
 
 
+def get_instance_id(
+    iid_file: str = CLOUDINIT_INSTANCE_ID_FILE
+) -> "Optional[str]":
+    """Query cloud instance-id from cmdline or CLOUDINIT_INSTANCE_ID_FILE"""
+    if util.which("cloud-id"):
+        # Present in cloud-init on >= Xenial
+        out, _err = util.subp(["cloud-init", "query", "instance_id"])
+        return out.strip()
+    if os.path.exists(iid_file):
+        return util.load_file(iid_file)
+    else:
+        logging.warning(
+            "Unable to determine current instance-id from %s" % iid_file
```

Comment:
No need to perform the substitution in logging calls:

```suggestion
            "Unable to determine current instance-id from %s", iid_file
```

Reply:
done.